src/plugins/files/server/file_client/file_metadata_client/adapters/es_index.ts

```diff
@@ -120,6 +121,10 @@ export class EsIndexFilesMetadataClient<M = unknown> implements FileMetadataClie
     };
   }
 
+  async bulkGet({ ids }: BulkGetArg): Promise<FileDescriptor[]> {
+    return await Promise.all(ids.map((id) => this.get({ id })));
```

Comment:
I don't believe cases will be using this adapter but could this use elasticsearch's mget API?

If that's not an option, we've used p-map in cases to limit the number of concurrent requests: https://github.com/sindresorhus/p-map